mk/spksrc.service.installer.dsm7

```diff
@@ -215,10 +230,12 @@ preupgrade ()
     call_func "validate_preupgrade"
 
     # dsm6 -> dsm7
-    # Migrate data to permanent storage
+    # Backup & Migrate data once to permanent storage
     if [ -d ${SYNOPKG_PKGDEST}/var -a ! "$(ls -A ${SYNOPKG_PKGVAR})" ]; then
-        # only migrate when destination is empty
-        $CP ${SYNOPKG_PKGDEST}/var/. ${SYNOPKG_PKGVAR} 2>&1 | install_log
+        echo "$RSYNC --ignore-existing ${SYNOPKG_PKGDEST}/var/ ${SYNOPKG_PKGVAR}" | install_log
+        $RSYNC --ignore-existing ${SYNOPKG_PKGDEST}/var/ ${SYNOPKG_PKGVAR} 2>&1 | install_log && chk=$?
+        $TAR -C ${SYNOPKG_PKGDEST} -Jcf - var > ${SYNOPKG_PKGVAR}/var-preDSM7-$(date +%Y%m%d-%H%M).tar.xz | install_log && chk="${chk}$?"
```

Comment:
What about any automated way to remove old backups? We don't want that file around forever.

Reply:
I am convinced that it "can" help when s*** hits the fan on complex packages.
In theory there would ever only be 1x backup file, kept for always...
Something I could add is to move the backup file back to target at postinst.
This assumes that it would get deleted when the user deletes or upgrade the package.
@publicarray any thoughts?

Reply:
@publicarray also, as rsync is using `--ignore-existing --remove-source-files` really there shouldn't be any files left.
That TAR backup could be removed entirely as there is nothing left to backup.